swarm/relay_check.py

```diff
@@ -0,0 +1,72 @@
+import aiohttp
+import asyncio
+from .util import load_json_file
+import os
+from .connection.connection import NodeWSConnection
+from .protocol.csm import CSM
+async def get_whitelisted_relays(config):
+    """Fetch whitelisted relays from the allowlist contract"""
+    async with NodeWSConnection(config['rpc']['execution_address']) as con:
+        contract = con.get_contract(
+            address=config['relay_allowlist_address'],
+            abi=load_json_file(os.path.join(os.getcwd(), 'abis', 'relay_allowlist.json'))
+        )
+        relays = await contract.functions.get_relays().call()
+        return relays
+
+async def check_validator_registration(session, relay_url, pubkey):
+    """Check if a validator is registered with a specific relay"""
+    url = f"{relay_url}/relay/v1/data/validator_registration"
+    try:
+        async with session.get(url, params={'pubkey': pubkey}) as response:
+            if response.status == 200:
+                return True
+            return False
+    except:
+        return False
```

Comment:
Maybe print the exception and that the result is not accurate because the GET failed.

Reply:
great point - thx